### ISSUE TITLE
please add examples of how to eager load nested entities:

### ISSUE THREAD

**Issue by samzilverberg** [skinny-framework/skinny-framework#428]
from
http://skinny-framework.org/documentation/orm.html#eager-loading :

> Note: eager loading of nested entities is not supported yet.

we're having a strugle with this atm and we're wondering how/if we could solve this with the skinny-orm classes (without going to "lower-level" scalajdbc or sql).

given a chained one-to-one relation:

Order (via order.user_id)-> 
User (belongs to one via user.group_id) -> 
Group (has "something" via Something.group_id) -> 
Something (here thers a group_id)

can you show an example of how we could load Order so we'd have access to

1. order.user.group
2. order.user.group.something

**Comment by samzilverberg** [comment]
update: after fiddling with this some more were able to achieve 1 with a combination of includes at the model level and at the model usage level.
it unfortunately creates an unnecessary extra query. 

the example with member and company shows this partially, but is a little harder and confusing to understand because company contains members (would be easier if it contained seq of something else entirely).

I'll try to comment with a suggestion docu improvment later. am writing from mobile phone atm

**Comment by samzilverberg** [comment]
this combination solved order.user.group loading:
- defining a BelongTo[User] on Order: `lazy val userRef = BelongsTo[User]...`
- chaining it with .includes[User]
- when fetching orders: `Order.includes(Order.userRef)`

itheres something funny about this.
if all BelongsTo are marked with `byDefault` (order.user & user.group), then the fetch query generated includes all the table joins and has all the data needed for extraction of all models.
BUT order.user.group is not extracted, and will only be extracted with the combination of includes - which then generates a second query to load users joined with their groups.

this seems a little suboptimal.
am i missing some way to do this with just 1 query and some other skinny orm methods?

If its too hard to follow I can try to create a mini project to demonstrate this. let me know

**Comment by seratch** [comment]
>order.user.group

As you mentioned using `#includes` and issuing one additional query for eager loading works.

>order.user.group.something

There is no way to fetch `something` except for manually running one more query like `select * from something where id in (${somethingIds})` and also returning a new list of `order.user.group` with manually attached `something` entities.

**Comment by samzilverberg** [comment]
just made a PR with an example of 1st level nested entity loading.
let me know if you find it ok, and of what you'd like to have changed on the PR.

I guess I bumped into the limitations of the orm with this issue, which is 100% fine imo. 
every ORM has its limitations (and quirks sometimes).
sometimes one just has to run more "lower level" sql 👍  (or sqls from scalajdbc)

I had a thought regarding this last bit.
Do you think it would be use `sqls` and extract from the resultset into my case classes using the extractors I've already written for all of my `object MyStuff extends SkinnyCRUDMapper[MyStuff]` entities ?
maybe you have an example of this already?

(I will RTFM of both libs if you dont have an example somewhere already :) )

**Comment by seratch** [comment]
Thanks for the PR for docs. Regarding your question here, understanding scalikejdbc (which is the foundation of Skinny ORM) may be helpful.

**Comment by samzilverberg** [comment]
your'e welcome.
we've read more into scalajdbc and are now using it much more often.

we knew all along that the ORM can't do it all and that at some point we'll startwriting some "lower level" queries. was just hoping to postpone the inevitable 😆
